Add crash rate analysis

poiru opened the pull request:
Some crash pings don't have `clientId` at all so I excluded those pings from the 'not first ping' check (i.e. they are treated as if they are not the first ping).

vitillo commented:
Please add a comment at the beginning of the notebook that explains what the analysis is about with a link to the corresponding BR in Bugzilla.

vitillo commented:
We shouldn't remove the first crash ping for a client as it might be associated with the N-th session of a client, where N > 0. What we should do instead is to consider only crash pings that were generated by the second, or greater, session.

vitillo commented:
The axes of the final plot should be labeled and the plot should have a legend.

vitillo commented:
There could be few individuals with frequent crashes that skew the results and we should make sure that this is not the case. Plotting the distribution of number of crashes per client for e10s and compare it to the one for non-e10s is going to help with that.

poiru commented:
@vitillo I addressed the review comments.

poiru commented:
@vitillo I addressed the review comments (see: [e10s_crash_rate.ipynb](http://nbviewer.ipython.org/github/vitillo/e10s_analyses/blob/7f772cf920bf833065cfbdcebe81e9ca3cfc3aeb/aurora/e10s_crash_rate.ipynb)). Note that I ran this with `fraction=0.1`. I'll run it with `fraction=1` if you're happy with it.

Vladan requested a few other things in the bug, but I'll leave that for a separate PR.

vitillo commented:
@poiru Where did you address my second comment?

poiru commented:
@vitillo This should address the 2nd comment.

vitillo commented:
"after/before the second session ping" should be "after/before the first session ping"

vitillo commented:
You should use first_session_ping_of_client, since you have to compute it anyway, to determine the creationTimestamp of the first session. Right now you are considering only crashes after the second session ping, not after first one as it should be.

poiru commented:
@vitillo I see, fixed.

vitillo commented:
The number of discarded crash pings seems pretty high. Could you please print the numbers of discarded crash pings due to missing clientId and discarded due to being generated before the first session ping?

poiru commented:
@vitillo Done. Most of the discarded pings were due to being generated before the first session.
